src/app/GitUI/UserControls/FileStatusList.cs

```diff
@@ -1257,268 +1182,175 @@ private void UpdateFileStatusListView(IReadOnlyList<FileStatusWithDescription> i
             }
 
             FileStatusListView.EndUpdate();
-            UpdateColumnWidth();
             return;
 
             void EnsureSelectedIndexChangeSubscription()
             {
                 _selectedIndexChangeSubscription ??= Observable.FromEventPattern(
-                        h => FileStatusListView.SelectedIndexChanged += h,
-                        h => FileStatusListView.SelectedIndexChanged -= h)
+                        h => FileStatusListView.FocusedNodeChanged += h,
+                        h => FileStatusListView.FocusedNodeChanged -= h)
                     .Where(x => _enableSelectedIndexChangeEvent)
                     .Throttle(SelectedIndexChangeThrottleDuration, MainThreadScheduler.Instance)
                     .ObserveOn(MainThreadScheduler.Instance)
                     .Subscribe(_ => FileStatusListView_SelectedIndexChanged());
             }
 
-            static int GetItemImageIndex(GitItemStatus gitItemStatus)
+            static string GetGroupName(FileStatusWithDescription i, int shownCount)
             {
-                string imageKey = GetItemImageKey(gitItemStatus);
+                bool hasRangeDiff = i.Statuses.Count > 0 && i.Statuses[^1].IsRangeDiff;
+                if (hasRangeDiff && i.Statuses.Count == 1)
+                {
+                    return i.Summary;
+                }
+
+                // Show shown and total number of files only if different; avoid showing "1/0" for "- No changes -"
+                string shownDisplay = shownCount >= i.Statuses.Count ? "" : $"{shownCount}/";
+                string rangeDiffDisplay = hasRangeDiff ? ",r" : "";
```

Comment:
I suggest this info is removed from the header, the number is the changed files.
Maybe even add RangeDiff at the top now?
Some tricky checks can be eliminated...